### DISCUSSION TITLE
Support for HS256 or RS256 signing algorithms?

### DISCUSSION THREAD

**Discussion by ghost** [nextauthjs/next-auth#727] [Help]
Hi,

I'm trying to use Next-Auth to authenticate users in MongoDB Realm. Realm requires either RS256 or HS256 signing algorithms. I'm trying to override the default HS512 with the following:

 ```
   encode: async ({ secret, token, maxAge }) => {
      const signingOptions = { algorithm: 'HS256' };
      const encodedToken = jwt.encode({ secret, token, maxAge, signingOptions });
      return encodedToken;
    },
    decode: async ({ secret, token, maxAge }) => {
      const verificationOptions = { algorithms: ['RS256', 'HS256', 'RS512', 'HS512'] };
      const verify = jwt.decode({ secret, token, maxAge, verificationOptions });
      return verify;
    }
```

I'm getting the following error: 
```
[next-auth][error][jwt_session_error] JWKKeySupport: the key does not support HS256 sign algorithm
```

Despite the fact that jose appears to support HS256 and RS256, according to the compatibility lists [here](https://jwt.io/) So, I assume I'm doing something wrong.

I saw a comment to a commit in July stating that tutorials on some of these options were forthcoming, but I couldn't find any such tutorials on the next-auth site. If there's a link somewhere explaining how to properly configure the signing algorithm, I'd appreciate being pointed to it.

**Comment by jotaRenan** [comment]
Hi @b-gibbs, were you able to find a solution to this issue?

**Reply by rijk** [reply]
Did you? :)
